model/src/main/scala/org/broadinstitute/dsde/rawls/model/WorkspaceModel.scala

```diff
@@ -187,11 +190,70 @@ case class EntityCopyDefinition(
                    entityNames: Seq[String]
                    )
 
+// In the next phase (GAWB-3100), this class's members will be swapped out for a singular methodUri
 case class MethodRepoMethod(
                    methodNamespace: String,
                    methodName: String,
                    methodVersion: Int
-                   )
+                   ) {
+  def asAgoraMethodUrl: String = asMethodUrlForRepo(MethodRepos.Agora.toString)
+
+  // Next phase: this method goes away
+  def asMethodUrlForRepo(repository: String): String = {
+
+    (existsAndNonEmpty(repository), MethodRepos.withName(repository), this.validate) match {
+      case (true, Some(_), Some(_)) => s"$repository://$methodNamespace/$methodName/$methodVersion"
+      case _ => throw new RawlsException(
+        s"Could not generate a method URI from MethodRepoMethod with repo \'$repository\', namespace \'$methodNamespace\', name \'$methodName\', version \'$methodVersion\'"
+      )
+    }
+  }
+
+  def validate: Option[MethodRepoMethod] = {
+    (existsAndNonEmpty(methodNamespace),
+     existsAndNonEmpty(methodName),
+     methodVersion > 0) match {
+      case (true, true, true) => Some(this)
+      case _ => None
+    }
+  }
+
+  private def existsAndNonEmpty(subject: String) = subject != null && subject.nonEmpty
```

Comment:
I still don't think you can get `null` here, so it's not really worth checking for. (I hate nulls in Scala, so try to avoid them wherever possible)